src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -400,7 +426,7 @@ public BibEntry resolveForStrings(BibEntry entry, boolean inPlace) {
      * care not to follow a circular reference pattern.
      * If the string is undefined, returns null.
      */
-    private String resolveString(String label, Set<String> usedIds) {
+    private String resolveString(String label, Set<String> usedIds, Set<String> allUsedIds) {
```

Comment:
Please add `Objects.requireNonNull(label);`, `Objects.requireNonNull(usedIds);`, `Objects.requireNonNull(allUsedIds);`

Please add `@paramter allUsedIds` containing a short description of the parameter